src/Spreadsheet/DeliverySpreadsheetParser.php

```diff
@@ -241,6 +248,24 @@ private function applyTags(TaggableInterface $task, $tagsAsString)
         }
     }
 
+    private function setTour(Delivery $delivery, string $routeName) {
+        foreach ($delivery->getTasks() as $task) {
+
+            $date = $task->getAfter();
+            $tour = $this->tourRepository->findByNameAndDate($routeName, $date);
+
+            if (is_null($tour)) {
+                $tour = new Tour();
+                $tour->setName($routeName);
+                $tour->setDate($date);
+                $this->entityManager->persist($tour);
```

Comment:
I'm not sure if it's a good place to do that. In my understanding, the responsibility of this class is parsing only, and persistence should be handled elsewhere (ImportDeliveriesHandler)

Reply:
yes that the main problem I have with this code, it looks like an unwanted side-effect...

I had difficulties to "find the tour" without persisting it, i ll try to do better

Reply:
see https://github.com/coopcycle/coopcycle-web/pull/4916/commits/c8d54d5053bb096e8d46333361788173b68f6b4d

Reply:
that looks better; the only thing: I would not 'pollute' the Delivery entity with `tourName` as that value is only needed during parsing. May a parser could return an object/array like this instead?

```
if (!$parseResult->rowHasErrors($rowNumber)) {
                $parseResult->addData($rowNumber, [
                    'delivery' => $delivery,
                    'tourName' => $record['tour.name'] ?? null
                ]);
            }
```

Reply:
I am not 100% convinced by this approach.

Maybe the good solution would have to be a DTO for "Delivery" that could be used for every input type or several DTO as DeliveryInput and childs CSVDeliveryInput, APIDeliveryInput ?